### PR TITLE
feat: first streak popup

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes
- Popup to show once the user has reached the first streak.
- As mentioned in the ticket, connecting with live data/mutation should be part of another ticket.

Preview:

Screenshot 2024-01-15 at 4 50 08 PM

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

MI-12 #done

### CODE REVIEW

File: packages/shared/src/components/modals/common.tsx
Comment (rebelchris) [inline packages/shared/src/components/modals/common.tsx:102]:
```suggestion
      /* webpackChunkName: "firstStreakModal" */ './streaks/FirstStreakModal'
```

File: packages/shared/src/components/modals/streaks/FirstStreakModal.tsx
Comment (rebelchris) [inline packages/shared/src/components/modals/streaks/FirstStreakModal.tsx:18]: This seems odd, shouldn't it just maintain modal guidelines?

File: packages/shared/src/components/modals/streaks/FirstStreakModal.tsx
Comment (rebelchris) [inline packages/shared/src/components/modals/streaks/FirstStreakModal.tsx:20]: Think you need absolute here as well with buttonV2 changes.
@kopancek ?

File: packages/shared/src/components/streak/popup/DayStreak.tsx
Comment (rebelchris) [inline packages/shared/src/components/streak/popup/DayStreak.tsx:49]: Should we maybe not use this as a Icon in this case?
Because it feels like we are abusing icon props to make it appear as one now?

**Review by rebelchris — COMMENTED** [review]
Nice some minor comments

**Review by DragosIorgulescu — APPROVED** [review]
Looks good overall, I think once existing comments are resolved this should be good to go

File: packages/shared/src/components/modals/streaks/FirstStreakModal.tsx
Comment (sshanzel) [reply packages/shared/src/components/modals/streaks/FirstStreakModal.tsx:18]: There was actually no padding probably because I did not use the Modal.Body. Let me check if I should actually be using them.

File: packages/shared/src/components/streak/popup/DayStreak.tsx
Comment (sshanzel) [reply packages/shared/src/components/streak/popup/DayStreak.tsx:49]: But this is an actual icon that renders an svg file.

File: packages/shared/src/components/streak/popup/DayStreak.tsx
Comment (sshanzel) [reply packages/shared/src/components/streak/popup/DayStreak.tsx:52]: Probably the primary issue is this, I felt like we don't have to turn this into an icon when we can simply create this as a div - and utilized the css just to make them consistent.

File: packages/shared/src/components/modals/streaks/FirstStreakModal.tsx
Comment (sshanzel) [reply packages/shared/src/components/modals/streaks/FirstStreakModal.tsx:20]: Moved to ModalClose. It should have a default `absolute` value.

File: packages/shared/src/components/streak/popup/DayStreak.tsx
Comment (rebelchris) [reply packages/shared/src/components/streak/popup/DayStreak.tsx:52]: Hmm just feels like it's a bit abusing things to me.
But if everyone else is happy not a blocker for me.

**Review by rebelchris — APPROVED** [review]
Non blocking left.

File: packages/shared/src/components/streak/popup/DayStreak.tsx
Comment (sshanzel) [reply packages/shared/src/components/streak/popup/DayStreak.tsx:52]: Yeah, totally understandable. Thoughts before merging? @dailydotdev/web-team - the image below is the elements we are referring to.
Screenshot 2024-01-16 at 4 54 58 PM

Comment (sshanzel) [reply packages/shared/src/components/streak/popup/DayStreak.tsx:52]: Merging this to unblock. Will then open a new PR if we find an alternative.

Comment (nensidosari) [reply packages/shared/src/components/streak/popup/DayStreak.tsx:52]: Personally think this should be a styled div...

**Review by nensidosari — APPROVED** [review]
On the Icon/Div discussions, I personally think that should be a simple styled Div rather then an icon, but non-blocking if people think it's best to be an Icon

**Pull request merged**
